server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -478,22 +478,27 @@ public long requireBuffer(
         throw new NoBufferForHugePartitionException(errorMessage);
       }
     }
-    return requireBuffer(requireSize);
+    return requireBuffer(appId, requireSize);
   }
 
-  public long requireBuffer(int requireSize) {
+  public long requireBuffer(String appId, int requireSize) {
     if (shuffleBufferManager.requireMemory(requireSize, true)) {
       long requireId = requireBufferId.incrementAndGet();
       requireBufferIds.put(
           requireId,
-          new PreAllocatedBufferInfo(requireId, System.currentTimeMillis(), requireSize));
+          new PreAllocatedBufferInfo(appId, requireId, System.currentTimeMillis(), requireSize));
       return requireId;
     } else {
       LOG.error("Failed to require buffer, require size: {}", requireSize);
       throw new NoBufferException("No Buffer For Regular Partition, requireSize: " + requireSize);
     }
   }
 
+  public long requireBuffer(int requireSize) {
+    // appId of EMPTY means the client uses the old version that should be upgraded.
```

Comment:
Could you elaborate more?

Reply:
Miss something?

Reply:
The appId of requireBuffer proto is introduce in #471 . So if using the older version client, the appId may be empty.

Reply:
OK. Maybe we should add a `todo` for this.

Reply:
But I found the current server has been not compatible with the version of #471 merged. That means we could remove this currently, let's do this in other PR.